File: php/huifier.php
<?php

function huify($word) {
    $word = mb_strtolower(trim($word));
    $vowels = 'аеёиоуыэюя';
    $rules = [
        'а' => 'я',
        'о' => 'ё',
        'у' => 'ю',
        'ы' => 'и',
        'э' => 'е',
    ];
    while (mb_strlen($word)) {
        $letter = mb_substr($word, 0, 1);
        if (mb_strpos($vowels, $letter) !== false) {
            if (array_key_exists($letter, $rules)) {
                $word = $rules[$letter] . mb_substr($word, 1);
            }
            break;
        } else {
            $word = mb_substr($word, 1);
        }
    }
    return empty($word) ? 'Хуй' : 'Ху'. $word;
}

File: go/huifier.go
package huifier

import (
    "strings"
    "unicode/utf8"
)

func removeFirstLetter(word string) string {
    _, i := utf8.DecodeRuneInString(word)
    return word[i:]
}

func Huify(word string) string {
    word = strings.ToLower(strings.TrimSpace(word))
    vowels := "аеёиоуыэюя"
    rules := map[rune]rune{
        'а': 'я',
        'о': 'е',
        'у': 'ю',
        'ы': 'и',
        'э': 'е',
    }
    for _, letter := range word {
        if strings.Contains(vowels, string(letter)) {
            if letter, ok := rules[letter]; ok {
                word = string(letter) + removeFirstLetter(word)
            }
            break
        } else {
            word = removeFirstLetter(word)
        }
    }
    if word != "" {
        return "Ху" + word
    } else {
        return "Хуй"
    }
}
